lib/mapUtils.js

```diff
@@ -292,3 +298,131 @@ export function openDirections(latitude, longitude, storeName) {
     alwaysIncludeGoogle: true,
   });
 }
+
+export function findStoreDistance(region, store) {
+  const distanceMeters = getDistance(region, store);
+  // Convert distance to 'x.xx' form, in miles units
+  const distance = convertDistance(distanceMeters, 'mi').toFixed(2);
+  return distance;
+}
+
+export async function orderStoresByDistance(region, inputStores) {
+  const sortedStores = [];
+  // Display distance in the StoreList
+  inputStores.forEach((store) => {
+    const currStore = store;
+    currStore.distance = findStoreDistance(region, store);
+    sortedStores.push(currStore);
+  });
+  // sorts in place
+  sortedStores.sort(function compare(a, b) {
+    return a.distance - b.distance;
+  });
+  return sortedStores;
+}
+
+export function findDefaultStore(stores) {
+  const defaultStore = stores.find((store) => {
+    return store.id === RecordIds.defaultStoreId;
+  });
+
+  const defaultRegion = {
+    latitude: defaultStore.latitude,
+    longitude: defaultStore.longitude,
+    ...deltas,
+  };
+  return { defaultStore, defaultRegion };
+}
+
+export function useCurrentLocation(setInitialLocation, setLocationPermissions) {
```

Comment:
Instead of passing through these state setters as callbacks, why not include these as part of the hook itself? You're already asking for location permissions in the hook, so I think you should consider having the hook return:

```javascript
const { loading, location, permissionStatus } = useCurrentLocation();
```

You want the statefulness of the location/location permissions status to be coupled with the hook itself (so you can extract it all instead of creating new stateful values every time you use the hook), unless I'm missing something?

Reply:
I _think_ I fixed this now? But wow as soon as I saw this something finally clicked and I realized I had this massive gap in my understanding of hooks :'))